qcodes/dataset/descriptions.py

```diff
@@ -0,0 +1,67 @@
+import io
+from typing import Dict, Any
+
+from ruamel.yaml import YAML
+
+from qcodes.dataset.dependencies import InterDependencies
+
+
+class RunDescriber:
+    """
+    The object that holds the description of each run in the database. This
+    object serialises itself to a string and is found under the run_description
+    column in the runs table
+
+    Extension of this object is planned for the future, for now it holds the
+    parameter interdependencies
```

Comment:
it seems that the attributes of this class which are included into the serialized version have to conform to a certain interface: having `serialize` and `deserialize` methods that perform the "this class" <-> dict transformation. Can this be mentioned somewhere? if such an interface is not implemented in code as an abstract base class (or any other "code"-way of mentioning this information), then at least shall this information land in some docstring or comment?

Reply:
Good point. For now, docstring. For the first extension, we can think about refactoring.